pkg/operator/k8sutil/deployment.go

```diff
@@ -69,22 +69,16 @@ func UpdateDeploymentAndWait(clusterContext *clusterd.Context, modifiedDeploymen
 	}
 
 	// Check whether the current deployment and newly generated one are identical
-	patchChanged := false
 	patchResult, err := patch.DefaultPatchMaker.Calculate(currentDeployment, modifiedDeployment)
 	if err != nil {
 		logger.Warningf("failed to calculate diff between current deployment %q and newly generated one. Assuming it changed. %v", currentDeployment.Name, err)
-		patchChanged = true
-	} else if !patchResult.IsEmpty() {
-		patchChanged = true
-	}
-
-	if !patchChanged {
+	} else if patchResult.IsEmpty() {
 		logger.Infof("deployment %q did not change, nothing to update", currentDeployment.Name)
 		return nil
 	}
 
 	// If deployments are different, let's update!
-	logger.Infof("updating deployment %q after verifying it is safe to stop", modifiedDeployment.Name)
+	logger.Infof("updating deployment %q after verifying it is safe to stop because this difference: %s", modifiedDeployment.Name, string(patchResult.Patch))
```

Comment:
This seems like it could be very verbose during an expected upgrade. What if we show the diff in a debug log message?

Reply:
Actually, the goal was to have it as Info so we can get some clarity more easily. But agreed it will be verbose during upgrades...

Reply:
Do you have an example for how verbose it will be for each osd that is updated?

Reply:
Probably large, but we don't upgrade every day :), I have mixed feelings too.

Reply:
How about this... If the ceph image changes we don't print the diff because it's obvious that it's an upgrade, but if the ceph image didn't change we could print the diff

Reply:
I need to see if we still have the `isUpgrade` flag, then certainly.

Reply:
We have `isUpgrade` but we cannot call it from this code :/